src/perl/Permabit/ModuleBase.pm

```diff
@@ -104,11 +106,22 @@ sub loadFromFiles {
   my $machine = $self->{machine};
   my $modFileName = $self->{modFileName};
   my $modVer = $self->{modVersion};
+  my $useDnf = $self->{useDnf};
 
   $machine->sendCommand("uname -m");
   my $arch = $machine->getStdout();
   chomp($arch);
 
+  # if useDnf(useLinuxNextModule), download vdo package from distribution repo
+  if ($useDnf) {
+    my $topdir = makeFullPath($machine->{workDir}, $self->{modVersion});
+    $self->_step(command => "mkdir -p $topdir");
+    my $getFromDnf = join(' ',
+                          "dnf", "download", "--destdir", "$self->{modDir}",
+                          "$modFileName"
+	                 );
+    $self->_step(command => $getFromDnf);
+  }
```

Comment:
This block seems specific to the user package. You're not getting the kernel module with dnf, are you? So this probably belongs in UserModule.pm instead.